src/network/networkd-link.c

```diff
@@ -2163,6 +2163,19 @@ static int link_set_can(Link *link) {
                         return log_link_error_errno(link, r, "Could not append IFLA_CAN_RESTART_MS attribute: %m");
         }
 
+        if (link->network->can_triple_sampling > -1) {
+                struct can_ctrlmode cm = { .mask = CAN_CTRLMODE_3_SAMPLES, .flags = 0 };
+
+                if (link->network->can_triple_sampling)
+                        cm.flags |= cm.mask;
```

Comment:
it might be nicer to write these all like this:

```c
struct can_ctrlmode cm = {
        .mask = CAN_CTRLMODE_3_SAMPLES,
        .flags = link->network->can_triple_sampling ?  CAN_CTRLMODE_3_SAMPLES : 0,
};
};

Reply:
Agreed, I'll change it.